packages/amplify-cli/src/plugin-manager.ts

```diff
@@ -24,7 +28,7 @@ export async function getPluginPlatform(): Promise<PluginPlatform> {
   // 3. re-scan if needed.
   // 4. write to update the plugins.json file if re-scan is performed
   // 5. return the pluginsInfo object
-  let pluginPlatform = readPluginsJsonFile();
+  pluginPlatform = pluginPlatform || readPluginsJsonFile();
```

Comment:
I'm actually concerned with this approach. Because in between the two method calls, the plugins.json file's content can be changed, and this will result in stale pluginPlatform being used. 
- if isNewCli() is called, and then based on the return a scan() is performed. 
- getPluginPlatform() method can result in scan() being called, which will write to the plugs.json file. Although in the current implementation, we don't have a scenario when isNewCli() called after getPluginPlatform(), but we don't have any mechanism to prevent such invocation sequence in the future, this can lead to unwary new engineers to make hard to detect bugs.

I'd suggest to remove the export of isNewCli(() and insert the postInstallation() directly into the getPlutinPlatform() method's logic. 
getPlutinPlatform() is designed with the intention to ensure the plugin platform is in a sound state, not just a simple read of the plugs.json file, that's why the scan() method is called in it.

Reply:
Okay I'm fine with making that change. I didn't want to add this logic to getPluginPlatform() because it seemed to be breaking single responsibility for the method, but if you think that's where it should go I'm fine with it

Reply:
I see your point, maybe we should rename the method getPlutinPlatform() then, considering its intended responsibility is more than just a "get".